### PR TITLE
Try It validations fixes

### PR DESCRIPTION
**Pull request by juancoen**
- Validate actual type for arrays (string[] will use string validation)
- Add object validator (will validate if string is valid JSON)
- TODO: add validation for union types

### CODE REVIEW

File: src/vendor/raml-validate/raml-validate.js [hunk 1]
```diff
@@ -57,6 +57,21 @@
   };
 
   /**
+   * Check if the value is a JSON string.
+   *
+   * @param  {String}  check
+   * @return {Boolean}
+   */
+  var isObject = function (check) {
```
Comment (xaka) [inline src/vendor/raml-validate/raml-validate.js]: Rename to `isJSON`?

File: src/vendor/raml-validate/raml-validate.js [hunk 2]
```diff
@@ -273,6 +288,23 @@
...
 
+  function isNativeType(typeName) {
+    typeName = typeName.replace('[]', '');
+    var nativeTypes = ['string', 'boolean', 'number', 'integer', 'object'];
+    return nativeTypes.indexOf(typeName) !== -1;
+  }
+
+  function convertType(config) {
+    if (Array.isArray(config.type)) {
+      config.type = config.type.map(function (aType) {
```
Comment (xaka) [inline src/vendor/raml-validate/raml-validate.js]: Is it fine to modify it in-place?

File: src/vendor/raml-validate/raml-validate.js (hunk 1 above)
Comment (juancoen) [inline src/vendor/raml-validate/raml-validate.js]: done

File: src/vendor/raml-validate/raml-validate.js (hunk 2 above)
Comment (juancoen) [inline src/vendor/raml-validate/raml-validate.js]: no, I cloned the config object so it doesn't get modified

**Comment by xaka** [comment]
LGTM :+1:

**Comment by m3l7** [comment]
I have a question on this:

many backend frameworks accepts object as a json in the body (i.e.  '{"id":3,"name":"test}'  )

or as query parameters ( i.e.     /request?id=3&name=test  )

which is quite different from current validation   (   /request?{"id":3,"name":"test"}   )

I know that this could lead to big problems with complex object (with array properties, nested strutures etc). I'm wondering if you have some suggestions/good practices on this problem.

**Comment by baoti** [comment]
Array is still not working.

```
/orders:
  get:
    queryParameters:
      tags:
        description: tags by array
        type: array
        required: false
        items:
          type: string
      tags2:
        description: tags by string[]
        type: string[]
        required: false

```

**Pull request merged**